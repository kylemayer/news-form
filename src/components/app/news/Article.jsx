import React from 'react';
import PropTypes from 'prop-types';

const Article = ({ title, author, description }) => (
  <div>
    <h4>{title}</h4>
    <p>{author}</p>
    <p>{description}</p>
  </div>
);

Article.propTypes = {
  title: PropTypes.string.isRequired,
  author: PropTypes.string,
  description: PropTypes.string,
};

export default Article;
