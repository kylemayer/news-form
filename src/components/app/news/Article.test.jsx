import dotenv from 'dotenv';
dotenv.config();

import React from 'react';
import { render, screen } from '@testing-library/react';
import Article from './Article';

describe('Article file', () => {
  it('should display an article object', async () => {
    render(
      <Article
        title="title"
        author="author"
        description="description"
      />);

    const title = await screen.findByText('title');
    const author = await screen.findByText('author');
    const descp = await screen.findByText('description');

    expect(title).toMatchSnapshot();
    expect(author).toMatchSnapshot();
    expect(descp).toMatchSnapshot();
  });
});
