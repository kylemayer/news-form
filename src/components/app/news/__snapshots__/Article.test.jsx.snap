// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article file should display an article object 1`] = `
<h4>
  title
</h4>
`;

exports[`Article file should display an article object 2`] = `
<p>
  author
</p>
`;

exports[`Article file should display an article object 3`] = `
<p>
  description
</p>
`;
